src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/SegmentAllocator.java

```diff
@@ -371,16 +409,17 @@ static SegmentAllocator arenaAllocator(long size, ResourceScope scope) {
      * The returned allocator might throw an {@link OutOfMemoryError} if an incoming allocation request exceeds
      * the system capacity.
      *
-     * @param scope the scope associated with the segments returned by this allocator.
+     * @param blockSize the block size associated with the arena-based allocator.
+     * @param scope the scope associated with the segments returned by the arena-based allocator.
      * @return a new unbounded arena-based allocator
      * @throws IllegalStateException if {@code scope} has been already closed, or if access occurs from a thread other
      * than the thread owning {@code scope}.
      */
-    static SegmentAllocator arenaAllocator(ResourceScope scope) {
+    static SegmentAllocator arenaUnbounded(long blockSize, ResourceScope scope) {
```

Comment:
Would be good to check if `blockSize > 0` here I think (and add javadoc, test for that).

Reply:
Will do